support/cas-server-support-gauth-mongo/src/test/java/org/apereo/cas/adaptors/gauth/MongoDbGoogleAuthenticatorTokenRepositoryTests.java

```diff
@@ -0,0 +1,243 @@
+                GoogleAuthenticatorMongoDbConfiguration.class,
+                CasCoreTicketsConfiguration.class,
+                CasCoreTicketCatalogConfiguration.class,
+                CasCoreLogoutConfiguration.class,
+                CasCoreHttpConfiguration.class,
+                CasCoreServicesConfiguration.class,
+                CasWebApplicationServiceFactoryConfiguration.class,
+                CasCoreAuthenticationConfiguration.class,
+                CasCoreServicesAuthenticationConfiguration.class,
+                CasCoreAuthenticationMetadataConfiguration.class,
+                CasCoreAuthenticationPolicyConfiguration.class,
+                CasCoreAuthenticationPrincipalConfiguration.class,
+                CasCoreAuthenticationHandlersConfiguration.class,
+                CasCoreAuthenticationSupportConfiguration.class,
+                CasPersonDirectoryConfiguration.class,
+                GoogleAuthenticatorAuthenticationEventExecutionPlanConfiguration.class,
+                AopAutoConfiguration.class,
+                CasCoreConfiguration.class,
+                CasCoreAuthenticationServiceSelectionStrategyConfiguration.class,
+                CasCoreUtilConfiguration.class,
+                RefreshAutoConfiguration.class,
+                CasCoreWebConfiguration.class})
+@EnableTransactionManagement(proxyTargetClass = true)
+@EnableAspectJAutoProxy(proxyTargetClass = true)
+@TestPropertySource(locations = {"classpath:/mongogauth.properties"})
+@EnableScheduling
+@ContextConfiguration(initializers = EnvironmentConversionServiceInitializer.class)
+public class MongoDbGoogleAuthenticatorTokenRepositoryTests {
+
+    @Autowired
+    @Qualifier("oneTimeTokenAuthenticatorTokenRepository")
+    private OneTimeTokenRepository repository;
+
+    @Test
+    public void verifyTokenSave() {
+        final GoogleAuthenticatorToken token = new GoogleAuthenticatorToken(1234, "casuser");
+        repository.store(token);
+        assertTrue(repository.exists("casuser", 1234));
+        assertTrue(token.getId() > 0);
+    }
+
+    @Test
+    public void verifyTokensWithUniqueIdsSave() {
+        final GoogleAuthenticatorToken token = new GoogleAuthenticatorToken(1234, "casuser");
+        repository.store(token);
+
+        final GoogleAuthenticatorToken token2 = new GoogleAuthenticatorToken(5678, "casuser");
+        repository.store(token2);
+        
+        assertTrue(token.getId() > 0);
+        assertTrue(token2.getId() > 0);
+        assertNotEquals(token.getId(), token2.getId());
+    }
+    
+    @TestConfiguration
+    public static class MongoTestConfiguration {
+        @Autowired
+        protected ApplicationContext applicationContext;
+
+        @PostConstruct
+        public void init() {
+            SchedulingUtils.prepScheduledAnnotationBeanPostProcessor(applicationContext);
+        }
+    }
+    
+}package org.apereo.cas.adaptors.gauth;
```

Comment:
It appears we have a copy/paste problem here.

Reply:
Oupss.

Sorry, I corrected.